src/components/CreateVmWizard/steps/Storage.js

```diff
@@ -0,0 +1,577 @@
+                      pullRight
+                    >
+                      <MenuItem
+                        id={`${kebabId}-edit`}
+                        onSelect={() => { this.inlineEditController.onActivate({ rowIndex, rowData }) }}
+                        disabled={actionsDisabled}
+                      >
+                        {msg.edit()}
+                      </MenuItem>
+                      <MenuItem
+                        id={`${kebabId}-delete`}
+                        onSelect={() => { this.onDeleteRow(rowData) }}
+                        disabled={actionsDisabled}
+                      >
+                        {msg.delete()}
+                      </MenuItem>
+                    </DropdownKebab>
+                  </Table.Cell>
+                }
+              </React.Fragment>
+            },
+          ],
+        },
+      },
+    ]
+  }
+
+  onCreateDisk () {
+    const newId = generateUnique('NEW_')
+
+    // If only 1 storage domain is available, select it automatically
+    const storageDomainList = createStorageDomainList(this.props.storageDomains, this.props.dataCenterId)
+    const storageDomainId = storageDomainList.length === 1 ? storageDomainList[0].id : '_'
+
+    // Setup a new disk in the editing hash
+    this.setState(state => ({
+      creating: newId,
+      editing: {
+        ...state.editing,
+        [newId]: {
+          id: newId,
+          name: suggestDiskName(this.props.vmName, this.props.disks),
+
+          diskId: '_',
+          storageDomainId,
+
+          bootable: false,
+          iface: 'virtio_scsi',
+          type: 'image',
+          format: getDefaultFormatType(this.props.optimizedFor),
+
+          size: 0,
+        },
+      },
+    }))
+  }
+
+  onEditDisk (rowData) {
+    this.setState(state => ({
+      editing: {
+        ...state.editing,
+        [rowData.id]: rowData,
+      },
+    }))
+  }
+
+  onDeleteRow (rowData) {
+    this.props.onUpdate({ remove: rowData.id })
+  }
+
+  handleCellChange (rowData, field, value) {
+    console.log('disk.id:', rowData.id, 'field:', field, 'value:', value)
+
+    if (field === 'size') {
+      if (isNumber(value) && value > 0 && value <= this.props.maxDiskSizeInGiB) {
+        value = +value * (1024 ** 3) // GiB to B
+      }
+    }
+
+    const editingRow = this.state.editing[rowData.id]
+    if (editingRow) {
+      editingRow[field] = value
+      this.setState(state => ({
+        editing: {
+          ...state.editing,
+          [rowData.id]: editingRow,
+        },
+      }))
+    }
+  }
+
+  // Push the new or editing row up via __onUpdate__
+  handleRowConfirmChange (rowData) {
+    const actionCreate = !!this.state.creating && this.state.creating === rowData.id
+    const editedRow = this.state.editing[rowData.id]
+
+    // TODO: Add field level validation for the edit or create fields
+
+    this.props.onUpdate({ [actionCreate ? 'create' : 'update']: editedRow })
```

Comment:
Same as previously, I think it will be better to move `actionCreate ? 'create' : 'update'` to separate constant, to make it more clear.

Reply:
That will just add clutter. Either way, using the object literal notation, it would still be { [updateType]: editedRow } which is no less confusing than how it is written now.